kingfisher_scrapy/extensions.py

```diff
@@ -288,3 +289,101 @@ def from_crawler(cls, crawler):
         extension = cls()
         sentry_sdk.init(sentry_dsn)
         return extension
+
+
+class KingfisherProcessNGAPI:
+    """
+    If the ``KINGFISHER_API_URI`` and ``KINGFISHER_API_KEY`` environment variables or configuration settings are set,
+    then messages are sent to a Kingfisher Process API for the ``item_scraped`` and ``spider_closed`` signals.
+    """
+    def __init__(self, url, username, password):
+        self.url = url
+        self.username = username
+        self.password = password
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        url = crawler.settings['KINGFISHER_NG_API_URI']
+        username = crawler.settings['KINGFISHER_NG_API_USERNAME']
+        password = crawler.settings['KINGFISHER_NG_API_PASSWORD']
+
+        if not url:
+            raise NotConfigured('KINGFISHER_NG_API_URI is not set.')
+
+        if (username and not password) or (password and not username):
+            raise NotConfigured('Both KINGFISHER_NG_API_USERNAME and KINGFISHER_NG_API_PASSWORD must be set.')
+
+        extension = cls(url, username, password)
+        crawler.signals.connect(extension.spider_opened, signal=signals.spider_opened)
+        crawler.signals.connect(extension.item_scraped, signal=signals.item_scraped)
+        crawler.signals.connect(extension.spider_closed, signal=signals.spider_closed)
+
+        return extension
+
+    def spider_opened(self, spider):
+        """
+        Sends an API request to start the collection in KP.
+        """
+        data = {
+            "source_id": spider.name,
+            "data_version": spider.get_start_time('%Y-%m-%d %H:%M:%S'),
+            "note": "collected by scrapy",
+            "sample": spider.sample,
+            "compile": True,
+            "upgrade": True,
+            "check": True,
+        }
+
+        response = self._post("api/v1/create_collection", data)
+
+        if not response.ok:
+            spider.logger.warning(
+                'Failed to POST create_collection. API status code: {}'.format(response.status_code))
+        else:
+            response_data = json.loads(response.text)
+            self.collection_id = response_data["collection_id"]
+
+    def spider_closed(self, spider, reason):
+        """
+        Sends an API request to close the collection.
+        """
+        data = {
+            "collection_id": self.collection_id,
+        }
+
+        response = self._post("api/v1/close_collection", data)
+
+        if not response.ok:
+            spider.logger.warning(
+                'Failed to post close collection. API status code: {}'.format(response.status_code))
+
+    def item_scraped(self, item, spider):
+        """
+        Sends an API request to store the file in Kingfisher Process.
+        """
+        if not item.get('post_to_api', True) or isinstance(item, PluckedItem):
```

Comment:
`post_to_api` has been gone a long time now, I think.

```suggestion
        if if isinstance(item, PluckedItem):
```